framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -428,20 +441,50 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 
 		// When # of CCMs are there on the outbox to be sent or # of blocks passed from last certified height
 		if (this._ccuFrequency >= newBlockHeader.height - this._lastCertifiedHeight) {
-			// TODO: _createCCU needs to be implemented which will create and send the CCU transaction
-
-			// @ts-expect-error unused constant ccmsWithHeight
 			const ccmsWithHeight = await this._sidechainChainConnectorStore.getCrossChainMessages();
 
-			// TODO: update `certificate` after 7908 is merged
-			const certificate = {} as Certificate;
-
-			// @ts-expect-error unused constant groupedCCMsBySize
-			const groupedCCMsBySize = this._groupCCMsBySize(
-				await this._sidechainChainConnectorStore.getCrossChainMessages(),
-				certificate,
+			const certificate = await this.getNextCertificateFromAggregateCommits(
+				this._lastCertifiedHeight,
+				aggregateCommits,
 			);
-			await this._createCCU();
+			if (!certificate) {
+				return;
+			}
+
+			const groupedCCMsBySize = this._groupCCMsBySize(ccmsWithHeight, certificate);
```

Comment:
since inboxUpdates is calculated based on the group of CCMs, [in PR we want to iterate over multiple inboxUpdates](https://github.com/LiskHQ/lisk-sdk/pull/7908/files#diff-09b22d0692c827e87992ce59088bb1b914c645389cf2c9aef77a7b9f3cee0435R374-R386) and call create CCU accordingly
So you can keep this part and we will update it in [PR](https://github.com/LiskHQ/lisk-sdk/pull/7908)

The idea is, we Iterate through InboxUpdates where if we only have one inboxUpdate then its simple that we take all the params calculated and then create a CCU. whereas when we have multiple inboxUpdates, (messageWitness is non empty and empty outboxRootWitness and certificate after first CCU for first inboxUpdate), we create multiple CCUs based on params needed for partial update